components/motor/gpio/motor_encoder.go

```diff
@@ -266,9 +263,9 @@ func (m *EncodedMotor) makeAdjustments(
 		newPowerPct = lastPowerPct
 	}
 
-	m.logger.Debugf("new powerPct %v", newPowerPct)
+	m.logger.CDebugf(ctx, "new powerPct %v", newPowerPct)
 
-	if err := m.setPower(ctx, newPowerPct, true); err != nil {
+	if err := m.setPower(ctx, newPowerPct); err != nil {
```

Comment:
need a lock here

Reply:
it's locked at the beginning of this func

Reply:
kk, my concern is `directionMovingInLock `has a sleep in it, so we are locking for longer than we need to. I see that the lock is used for both `directionMovingInLock `and `setPower`. 

I am wondering if this function should be using `DirectionMoving`, and if `DirectionMoving`/`directionMovingInLock` should be updated to only lock to access lastPowerPct.

Reply:
I see. I don't think it should affect rpmMonitor because we block rpmMonitor from switching directions, and that's the only time directionMoving sleeps.